internal/services/communication/communication_service_resource.go

```diff
@@ -146,7 +147,8 @@ func (r CommunicationServiceResource) Create() sdk.ResourceFunc {
 				// The location is always `global` from the Azure Portal
 				Location: location.Normalize("global"),
 				Properties: &communicationservices.CommunicationServiceProperties{
-					DataLocation: model.DataLocation,
+					DataLocation:  model.DataLocation,
+					LinkedDomains: pointer.To(model.LinkedDomains),
```

Comment:
LinkedDomains is an optional value, so `nil`  should be used. if the user does not provide `LinkedDomains`, model. LinkedDomains should be a nil value, since its an slice? Pretty much the same behavior as for tags here.

Reply:
the `pointer.To()` function won't return `nil` there, it will return an empty slice. If the value should be `nil` for an empty slice, then that will need to be conditionally set outside the construction of the `CommunicationServiceProperties` if `model.LinkedDomains` is not empty. If an empty slice there is treated the same way as `nil` by the API, then this is fine as-is.

Tags have a slightly different behaviour, and are a `map[string]string` (or similar, depending on service implementation), so are handled differently.

Reply:
if the `pointer.To()` function won't return nil there, could you please explain why the debugger shows me a nil value here? I'm running the `TestAccCommunicationService_basic` test here:

<img width="1409" alt="image" src="https://github.com/hashicorp/terraform-provider-azurerm/assets/1560587/619545ef-0431-4d15-aa8b-d6b6b30dd509">

I as know, `pointer.To()` just creates a pointer of an variable. If the end-user omits linked_domains, the `model.LinkedDomains` value is nil.

Could you please explain, why we need an additional condition here? I just want to get the reason behind.

Reply:
I added a test case for any empty list anyways which is passing.